src/lib/components/custom/theme-selector.svelte

```diff
@@ -9,6 +9,36 @@
 	// without this line the dropdown does not appear...
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars
 	const mode = $derived(theme.mode)
+	let invalidColor = $state(false)
+
+	function formatHexColor() {
+		let hex = theme.baseColor
+		hex = hex.replace(/^#/, '')
+
+		if (hex.length === 3) {
+			hex = hex.replace(/(.)/g, '$1$1')
+		}
+
+		return (theme.baseColor = '#' + hex)
+	}
+	function validateColor() {
+		const hexRegex = /^#?([a-f0-9]{6}|[a-f0-9]{3})$/i
+
+		if (!theme.baseColor.startsWith('#')) {
+			theme.baseColor = '#' + theme.baseColor
+		}
+		let timeout = setTimeout(() => {
+			if (theme.baseColor.length === 4) {
+				formatHexColor()
+			}
+		}, 3000)
```

Comment:
This should not happen here I think. It should be extended only in the function that updates the CSS and not saved in the theme store.